### ISSUE TITLE
When trying to use web3 on Windows 10 in Powershell or CMD, get "ModuleNotFoundError: No module named 'win32file'"

### ISSUE THREAD

**Issue by Mikerah** [ethereum/web3.py#454]
* Version: 3.16.2
* Python: 3.6
* OS: win10

### What was wrong?

When trying to call `web3.eth.filter('latest')` by following the example in the documentation, I get the following error:
~~~~
Python 3.6.3 (v3.6.3:2c5fed8, Oct  3 2017, 17:26:49) [MSC v.1900 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from web3 import Web3, IPCProvider
>>> web3 = Web3(IPCProvider())
>>> web3.eth.filter('latest')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\eth.py", line 270, in filter
    "eth_newBlockFilter", [],
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\manager.py", line 93, in request_blocking
    response = self._make_request(method, params)
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\manager.py", line 76, in _make_request
    return request_func(method, params)
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\middleware\attrdict.py", line 20, in middleware
    response = make_request(method, params)
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\middleware\formatting.py", line 25, in middleware
    response = make_request(method, params)
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\providers\ipc.py", line 84, in make_request
    with get_ipc_socket(self.ipc_path) as sock:
  File "c:\python36\Lib\contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\providers\ipc.py", line 25, in get_ipc_socket
    from web3.utils.windows import NamedPipe
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\utils\windows.py", line 8, in <module>
    import win32file  # noqa: E402
ModuleNotFoundError: No module named 'win32file'
~~~~

However, I do not get this message when I run the code from a forked version of the web3.py repository in either Powershell or CMD.

I have also attempted to run this with and without geth running in the background.

### How can it be fixed?

When checking if all the requirements for web3 are installed, I noticed that the library `pypiwin32` is not installed. But, when I checked the `setup.py` file for this repository, it is clearly there.

For reference, requirements for my project:
~~~~
certifi==2017.11.5
chardet==3.0.4
cytoolz==0.8.2
ethereum-abi-utils==0.4.4
ethereum-keys==0.1.0a7
ethereum-tester==0.1.0b2
ethereum-utils==0.6.0
idna==2.6
pylru==1.0.9
pymongo==3.5.1
pysha3==1.0.2
requests==2.18.4
rlp==0.6.0
semantic-version==2.6.0
toolz==0.8.2
urllib3==1.22
web3==3.16.2
~~~~

To try to resolve this, I installed the `pypiwin32` library myself. But, then I got this error message:
~~~~
Traceback (most recent call last):
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\utils\windows.py", line 17, in __init__
    0, None, win32file.OPEN_EXISTING, 0, None)
pywintypes.error: (2, 'CreateFile', 'The system cannot find the file specified.')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\eth.py", line 270, in filter
    "eth_newBlockFilter", [],
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\manager.py", line 93, in request_blocking
    response = self._make_request(method, params)
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\manager.py", line 76, in _make_request
    return request_func(method, params)
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\middleware\attrdict.py", line 20, in middleware
    response = make_request(method, params)
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\middleware\formatting.py", line 25, in middleware
    response = make_request(method, params)
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\providers\ipc.py", line 84, in make_request
    with get_ipc_socket(self.ipc_path) as sock:
  File "c:\python36\Lib\contextlib.py", line 81, in __enter__
    return next(self.gen)
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\providers\ipc.py", line 27, in get_ipc_socket
    pipe = NamedPipe(ipc_path)
  File "C:\Users\Mikerah\Documents\GitHub\py-eth-parser\env\lib\site-packages\web3\utils\windows.py", line 19, in __init__
    raise IOError(err)
OSError: (2, 'CreateFile', 'The system cannot find the file specified.')
~~~~

However, when I run geth in the background, this error is resolved.

Thus, in order to fix this, we need to:
1. Install `pypiwin32`
2. Specify in the documentation that you need to have geth installed.

**Comment by Mikerah** [comment]
I noticed, upon further investigation, that the version of this library that is uploaded to PyPi and the version that is currently in the repository are different. The version uploaded to PyPi was last updated on Oct 11. This maybe why I am experiencing some issues with using this library.

**Comment by carver** [comment]
Thanks for the report! Our primary devs aren't running Windows locally, so it's really helpful to get some feedback here, especially before #166 is complete.

> I noticed that the library `pypiwin32` is not installed

Indeed this is strange. Any chance that `sys.platform` is not `"win32"` on your machine?

> Specify in the documentation that you need to have geth installed.

Would you expand on this? Were you trying to use a different Ethereum than geth? Or were you expecting to run without any local client?

**Comment by Mikerah** [comment]
I checked `sys.platform` on my machine and it was `win32`. 

As for the second part, the documentation was unclear about whether I needed a client to run in the background or if web3.py started an instance of testrpc or if it needed anything at all to run.

When I tried to run the code on campus, I got an error that I can no longer reproduce when I am at home as it works. The error was `win32file.createFile “The System cannot find the specified path”`. Maybe, I simply needed to restart my instance of Powershell for the effects to occur.
